### ISSUE TITLE
Cannot start Embedded Payara in Arquillian tests

### ISSUE THREAD

**Issue by dmatej** [payara/Payara#2276]
# Description #
----------

Start of the container ends with NullpointerException.
```
SEVERE: [192.168.11.6]:4900 [development] [3.9.1] null
java.lang.NullPointerException
        at fish.payara.nucleus.hazelcast.DomainDiscoveryService.discoverNodes(DomainDiscoveryService.java:147)
        at com.hazelcast.internal.cluster.impl.DiscoveryJoiner.getPossibleAddresses(DiscoveryJoiner.java:70)
```

## Expected Outcome

Running container and tests. 
It worked with version 5.0.0.Alpha4-SNAPSHOT built in november 2017.

## Current Outcome
```
Led 16, 2018 10:55:25 DOP. com.sun.enterprise.v3.server.CommonClassLoaderServiceImpl findH2Client
INFO: Cannot find h2db client jar file, h2 jdbc driver will not be available by default.
Led 16, 2018 10:55:26 DOP. org.glassfish.security.services.impl.authorization.AuthorizationServiceImpl initialize
INFO: Authorization Service has successfully initialized.
10:55:26,138 INFO  [main]     internal.util.Version: HV000001: Hibernate Validator null
Led 16, 2018 10:55:27 DOP. com.sun.enterprise.v3.services.impl.GrizzlyProxy start0
INFO: Grizzly Framework 2.4.1 started in: 28ms - bound to [/0.0.0.0:58888]
Led 16, 2018 10:55:27 DOP. com.sun.enterprise.v3.services.impl.GrizzlyProxy start0
INFO: Grizzly Framework 2.4.1 started in: 6ms - bound to [/0.0.0.0:58889]
Led 16, 2018 10:55:27 DOP. com.sun.enterprise.v3.admin.adapter.AdminEndpointDecider setGuiContextRoot
INFO: Admin Console Adapter: context root: /admin
Led 16, 2018 10:55:27 DOP. com.sun.enterprise.v3.admin.adapter.AdminEndpointDecider setGuiContextRoot
INFO: Admin Console Adapter: context root: /admin
Led 16, 2018 10:55:27 DOP. com.sun.enterprise.v3.admin.adapter.AdminEndpointDecider setGuiContextRoot
INFO: Admin Console Adapter: context root: /admin
Led 16, 2018 10:55:27 DOP. com.hazelcast.instance.AddressPicker
INFO: [LOCAL] [development] [3.9.1] Using bind address: /192.168.11.6:4900
Led 16, 2018 10:55:27 DOP. com.hazelcast.instance.AddressPicker
INFO: [LOCAL] [development] [3.9.1] Using public address: /192.168.11.6:4900
Led 16, 2018 10:55:27 DOP. com.hazelcast.system
INFO: [192.168.11.6]:4900 [development] [3.9.1] Hazelcast 3.9.1 (20171130 - feca534) starting at [192.168.11.6]:4900
Led 16, 2018 10:55:27 DOP. com.hazelcast.system
INFO: [192.168.11.6]:4900 [development] [3.9.1] Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
Led 16, 2018 10:55:27 DOP. com.hazelcast.system
INFO: [192.168.11.6]:4900 [development] [3.9.1] Configured Hazelcast Serialization version: 1
Led 16, 2018 10:55:28 DOP. com.hazelcast.spi.impl.operationservice.impl.BackpressureRegulator
INFO: [192.168.11.6]:4900 [development] [3.9.1] Backpressure is disabled
Led 16, 2018 10:55:28 DOP. com.hazelcast.instance.Node
INFO: [192.168.11.6]:4900 [development] [3.9.1] Activating Discovery SPI Joiner
Led 16, 2018 10:55:28 DOP. com.hazelcast.spi.impl.operationexecutor.impl.OperationExecutorImpl
INFO: [192.168.11.6]:4900 [development] [3.9.1] Starting 4 partition threads and 3 generic threads (1 dedicated for priority tasks)
Led 16, 2018 10:55:28 DOP. com.hazelcast.internal.diagnostics.Diagnostics
INFO: [192.168.11.6]:4900 [development] [3.9.1] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
Led 16, 2018 10:55:28 DOP. com.hazelcast.core.LifecycleService
INFO: [192.168.11.6]:4900 [development] [3.9.1] [192.168.11.6]:4900 is STARTING
Led 16, 2018 10:55:28 DOP. com.hazelcast.internal.cluster.impl.DiscoveryJoiner
SEVERE: [192.168.11.6]:4900 [development] [3.9.1] null
java.lang.NullPointerException
        at fish.payara.nucleus.hazelcast.DomainDiscoveryService.discoverNodes(DomainDiscoveryService.java:147)
        at com.hazelcast.internal.cluster.impl.DiscoveryJoiner.getPossibleAddresses(DiscoveryJoiner.java:70)
        at com.hazelcast.internal.cluster.impl.DiscoveryJoiner.getPossibleAddressesForInitialJoin(DiscoveryJoiner.java:59)
        at com.hazelcast.cluster.impl.TcpIpJoiner.joinViaPossibleMembers(TcpIpJoiner.java:131)
        at com.hazelcast.cluster.impl.TcpIpJoiner.doJoin(TcpIpJoiner.java:90)
        at com.hazelcast.internal.cluster.impl.AbstractJoiner.join(AbstractJoiner.java:134)
        at com.hazelcast.instance.Node.join(Node.java:690)
        at com.hazelcast.instance.Node.start(Node.java:390)
        at com.hazelcast.instance.HazelcastInstanceImpl.<init>(HazelcastInstanceImpl.java:133)
        at com.hazelcast.instance.HazelcastInstanceFactory.constructHazelcastInstance(HazelcastInstanceFactory.java:195)
        at com.hazelcast.instance.HazelcastInstanceFactory.newHazelcastInstance(HazelcastInstanceFactory.java:174)
        at com.hazelcast.instance.HazelcastInstanceFactory.newHazelcastInstance(HazelcastInstanceFactory.java:124)
        at com.hazelcast.core.Hazelcast.newHazelcastInstance(Hazelcast.java:58)
        at fish.payara.nucleus.hazelcast.HazelcastCore.bootstrapHazelcast(HazelcastCore.java:447)
        at fish.payara.nucleus.hazelcast.HazelcastCore.event(HazelcastCore.java:258)
        at org.glassfish.kernel.event.EventsImpl.send(EventsImpl.java:131)
        at com.sun.enterprise.v3.server.ApplicationLoaderService.postConstruct(ApplicationLoaderService.java:331)
        at org.jvnet.hk2.internal.ClazzCreator.postConstructMe(ClazzCreator.java:327)
        at org.jvnet.hk2.internal.ClazzCreator.create(ClazzCreator.java:375)
        at org.jvnet.hk2.internal.SystemDescriptor.create(SystemDescriptor.java:487)
        at org.glassfish.hk2.runlevel.internal.AsyncRunLevelContext.findOrCreate(AsyncRunLevelContext.java:305)
        at org.glassfish.hk2.runlevel.RunLevelContext.findOrCreate(RunLevelContext.java:85)
        at org.jvnet.hk2.internal.Utilities.createService(Utilities.java:2126)
        at org.jvnet.hk2.internal.ServiceHandleImpl.getService(ServiceHandleImpl.java:116)
        at org.jvnet.hk2.internal.ServiceHandleImpl.getService(ServiceHandleImpl.java:90)
        at org.glassfish.hk2.runlevel.internal.CurrentTaskFuture$QueueRunner.oneJob(CurrentTaskFuture.java:1237)
        at org.glassfish.hk2.runlevel.internal.CurrentTaskFuture$QueueRunner.run(CurrentTaskFuture.java:1168)
        at org.glassfish.hk2.runlevel.internal.CurrentTaskFuture$UpOneLevel.run(CurrentTaskFuture.java:786)
        at org.glassfish.hk2.runlevel.internal.CurrentTaskFuture$UpAllTheWay.go(CurrentTaskFuture.java:585)
        at org.glassfish.hk2.runlevel.internal.CurrentTaskFuture$UpAllTheWay.access$100(CurrentTaskFuture.java:467)
        at org.glassfish.hk2.runlevel.internal.CurrentTaskFuture.go(CurrentTaskFuture.java:157)
        at org.glassfish.hk2.runlevel.internal.CurrentTaskFuture.go(CurrentTaskFuture.java:152)
        at org.glassfish.hk2.runlevel.internal.AsyncRunLevelContext.proceedTo(AsyncRunLevelContext.java:535)
        at org.glassfish.hk2.runlevel.internal.RunLevelControllerImpl.proceedTo(RunLevelControllerImpl.java:72)
        at com.sun.enterprise.v3.server.AppServerStartup.proceedTo(AppServerStartup.java:556)
        at com.sun.enterprise.v3.server.AppServerStartup.run(AppServerStartup.java:348)
        at com.sun.enterprise.v3.server.AppServerStartup.doStart(AppServerStartup.java:234)
        at com.sun.enterprise.v3.server.AppServerStartup.start(AppServerStartup.java:223)
        at com.sun.enterprise.glassfish.bootstrap.GlassFishImpl.start(GlassFishImpl.java:81)
        at fish.payara.arquillian.container.payara.embedded.PayaraContainer.start(PayaraContainer.java:194)
        at org.jboss.arquillian.container.impl.ContainerImpl.start(ContainerImpl.java:199)
...
```

## Steps to reproduce (Only for bug reports) 

mvn clean test

## Environment ##

- **Payara Version**: 5.Beta1, same with the commit b9add8a30db47 of Payara-5 branch
- **Edition**: payara-embedded-all
- **JDK Version**: Oracle JDK8u144
- **Operating System**: Linux
- **Database**: No.

**Comment by dmatej** [comment]
Simple test project:
```
git clone git@github.com:dmatej/java-ee-bug-reports.git
cd java-ee-bug-reports;
git checkout Payara-2276
mvn clean verify -Dpayara.version=5.0.0.Alpha4-SNAPSHOT; # success
mvn clean verify -Dpayara.version=5.Beta2-SNAPSHOT; # fail
mvn clean verify -Dpayara.version=5.Beta1; # fail
```

**Comment by smillidge** [comment]
thanks for the report I will try to reproduce. Embedded should not be starting Hazelcast so we likely missed some config somewhere

**Comment by dmatej** [comment]
I suspect this commit: 8f54e86db12

**Comment by smillidge** [comment]
Fixed by #2389
